exporters/otlp/CMakeLists.txt

```diff
@@ -113,8 +113,8 @@ if(WITH_OTLP_HTTP)
 
   target_link_libraries(
     opentelemetry_exporter_otlp_http_client
-    PUBLIC opentelemetry_sdk opentelemetry_proto opentelemetry_http_client_curl
```

Comment:
opentelemetry_exporter_otlp_http_client use the headers of opentelemetry_sdk and opentelemetry_ext.I think it's better to keep opentelemetry_sdk and opentelemetry_ext public and move opentelemetry_proto, opentelemetry_http_client_curl and nlohmann_json into private link.

Reply:
This makes sense - I think I lazily just made everything private instead of looking more carefully. Thanks for the review.